tests/sweep_framework/sweeps/matmul/short/matmul_user_program_config.py

```diff
@@ -0,0 +1,569 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+from loguru import logger
+import enum
+
+import torch
+
+import ttnn
+
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from models.utility_functions import torch_random
+
+
+# Set up suites and specify batch_sizes, input_shapes, batch_matrix_multiply, program_config, and memory configs
+# TODO: Missing coverage for WH fp32 and compute config tests in: tests/tt_eager/python_api_testing/unit_testing/misc/test_matmul.py
+# [1d in0 batched, bmm no mcast per_core_M > M, bmm no mcast per_core_M < M, 1d mcast in1 with height padding, 1d mcast in1, 1d mcast in0, 2d mcast, 2d mcast transposed, 2d mcast height/width sharded] x [specific L1 memory, dram memory] x [a=bfloat16, a=bfloat8_b] x [b=bfloat16, b=bfloat8_b] x [out=bfloat16, out=bfloat8_b]
+# 144 tests.
+# Use the first two categories for suites: 18 suites with 8 tests each.
+parameters = {
+    "matmul_1d_in0_batched": {
+        "batch_sizes": [(16,)],
+        "input_shapes": [(128, 128, 1024)],
+        "batch_matrix_multiply": [False],
+        "program_config": [
+            ttnn.MatmulMultiCoreReuseMultiCast1DProgramConfig(
```

Comment:
This ttnn.MatmulMultiCoreReuseMultiCase1DProgramConfig class does not have the serialization pybinds required in cpp. I'm adding validation to warn test writers if their types aren't pybinded with the serialization. ATM, we need to add these pybinds to use with tests.

Reply:
What do I need to do have the right serialization pybinds?

Are the other configs ok (in ttnn/cpp/ttnn/operations/matmul/matmul_pybind.hpp) or would I need to change all of them?

Reply:
Need to update pybinds to use tt_pybind_class as specified in ttnn/cpp/ttnn/operations/matmul/matmul_pybind.hpp

Reply:
Looks good now, thx.